SalesforceFoundation/EDA
# Fixed NPE and Added a New Unit Test

# Critical Changes

# Changes
We've fixed a null pointer exception that occurs when the record type of an Account with a primary affiliation is changed to a record type with no affiliated mapping under EDA Settings. With this change, EDA now updates the Account's record type as specified when this type of change occurs.


# Issues Closed
Fixes <a href="https://github.com/SalesforceFoundation/EDA/issues/820#issuecomment-499705630">#820</a>

# New Metadata

# Deleted Metadata

# Testing Notes
1. Create a Contact
2. Create an Account with Department record type - this record type should exist in the Affiliation Mappings in the EDA Settings
3. Create a Primary Affiliation between Account and Contact
4. Update Account record type to Business Organization - this record type should NOT exist in the Affiliation Mappings in the EDA Settings
5. System should not throw an exception and "Primary Department" field should be cleared.
